src/test/java/io/vlingo/wire/fdx/bidirectional/AbstractServerChannelActorTest.java

```diff
@@ -0,0 +1,187 @@
+// Copyright © 2012-2018 Vaughn Vernon. All rights reserved.
+//
+// This Source Code Form is subject to the terms of the
+// Mozilla Public License, v. 2.0. If a copy of the MPL
+// was not distributed with this file, You can obtain
+// one at https://mozilla.org/MPL/2.0/.
+package io.vlingo.wire.fdx.bidirectional;
+
+import io.vlingo.actors.World;
+import io.vlingo.wire.channel.RequestChannelConsumerProvider;
+import io.vlingo.wire.channel.ResponseChannelConsumer;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.junit.Assert.assertEquals;
+
+public abstract class AbstractServerChannelActorTest {
+  private static AtomicInteger TEST_PORT = new AtomicInteger(49560);
```

Comment:
Each test, extending from this class, should have their own port range to avoid collision when running parallel tests .

Reply:
Got it. Is it possible to use random ports on all tests, allowing a definitive solution for problems like this?

Reply:
Even a single static `AtomicInteger` and using `incrementAndGet()` in the abstract base will work for all extenders. I don't know why random would be better. You would still have to set ranges to different tests don't collide. I don't think there could be a central static because it could get recreated if multiple JVMs are employed in a given test run.

Reply:
When I say random I mean letting the OS choose a random local port and configuring the client accordingly. This is a common failsafe practice on testing server applications.

Reply:
I could not find any reliable way to do what you are suggesting with raw `java.nio.channels.SocketChannel` and `io.rsocket.transport.netty.server.TcpServerTransport` at the time. I think most of the frameworks simply loop through sockets until find one that's open. We could replicate the  approach but it's prone to race conditions when the loop identifies a free socket, closes it, and the consumer did not yet attached to it.

Reply:
This automatic port allocation is provided by every ServerSocket implementation. See https://docs.oracle.com/javase/8/docs/api/java/net/ServerSocket.html#ServerSocket-int-

Moreover, the test at https://github.com/rsocket/rsocket-java/blob/88ee909b8bfb78b5c53fa25bc39e16f6d893f0df/rsocket-transport-netty/src/test/java/io/rsocket/transport/netty/server/TcpServerTransportTest.java#L85 suggests that this mechanism is also available on rsocket.

Reply:
Great find! Can you test this?

Reply:
@alexguzun Can this be merged?